commit e7b8ac8397721f7870ccb2158275afe4a463878f
Author: Eike Hein <hein@kde.org>
Date:   Wed Aug 14 16:24:54 2013 +0200

    Add back window ids to the drag data payload.
    
    This e.g. makes dropping tasks on the pager work again.
    
    BUG:322567

diff --git a/libs/taskmanager/tasksmodel.cpp b/libs/taskmanager/tasksmodel.cpp
index b849528..472cd76 100644
--- a/libs/taskmanager/tasksmodel.cpp
+++ b/libs/taskmanager/tasksmodel.cpp
@@ -25,6 +25,7 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #include "tasksmodel.h"
 
 #include <QMetaEnum>
+#include <QMimeData>
 
 #include <KDebug>
 
@@ -171,6 +172,22 @@ QVariant TasksModel::data(const QModelIndex &index, int role) const
         }
 
         return windows;
+    } else if (role == TasksModel::MimeType) {
+        if (item->itemType() == TaskItemType) {
+            return Task::mimetype();
+        } else if (item->itemType() == GroupItemType) {
+            return Task::groupMimetype();
+        }
+    } else if (role == TasksModel::MimeData) {
+        QMimeData mimeData;
+
+        item->addMimeData(&mimeData);
+
+        if (item->itemType() == TaskItemType) {
+            return QVariant::fromValue(mimeData.data(Task::mimetype()));
+        } else if (item->itemType() == GroupItemType) {
+            return QVariant::fromValue(mimeData.data(Task::groupMimetype()));
+        }
     }
 
     return QVariant();
diff --git a/libs/taskmanager/tasksmodel.h b/libs/taskmanager/tasksmodel.h
index d87d037..9c1b3f7 100644
--- a/libs/taskmanager/tasksmodel.h
+++ b/libs/taskmanager/tasksmodel.h
@@ -63,7 +63,9 @@ public:
         Active = Qt::UserRole + 17,
         DemandsAttention = Qt::UserRole + 18,
         LauncherUrl = Qt::UserRole + 19,
-        WindowList = Qt::UserRole + 20
+        WindowList = Qt::UserRole + 20,
+        MimeType = Qt::UserRole + 21,
+        MimeData = Qt::UserRole + 22
     };
 
     explicit TasksModel(GroupManager *groupManager, QObject *parent = 0);
diff --git a/plasma/desktop/applets/tasks/package/contents/ui/Task.qml b/plasma/desktop/applets/tasks/package/contents/ui/Task.qml
index c0fa99b..9efbff5 100644
--- a/plasma/desktop/applets/tasks/package/contents/ui/Task.qml
+++ b/plasma/desktop/applets/tasks/package/contents/ui/Task.qml
@@ -56,6 +56,10 @@ DragArea {
         url: model.LauncherUrl
     }
 
+    onDragStarted: {
+        mimeData.setData(model.MimeType, model.MimeData);
+    }
+
     onItemIndexChanged: {
         if (!inPopup && !tasks.vertical && Layout.calculateStripes() > 1) {
             var newWidth = Layout.taskWidth();
